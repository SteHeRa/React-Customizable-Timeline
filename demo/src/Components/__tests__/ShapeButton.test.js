import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ShapeButton from '../ShapeButton/ShapeButton';

describe('ShapeButton', () => {
  it('component renders correctly', () => {
    const shapebutton = render(
      <ShapeButton
        shapes={{
          titleShape: 'circle',
          dotShape: 'circle',
          lineShape: 'solid',
        }}
        setShapes={setShapes}
      />,
    );
    expect(shapebutton).toMatchSnapshot();
  });

  it('calls when clicked', () => {});

  const setShapes = jest.fn();

  render(
    <ShapeButton
      shapes={{
        titleShape: 'circle',
        dotShape: 'circle',
        lineShape: 'solid',
      }}
      setShapes={setShapes}
    />,
  );

  for (let i = 0; i < 12; i++) {
    userEvent.click(screen.getAllByRole('radio')[i]);
    expect(setShapes).toHaveBeenCalledTimes(i + 1);
  }
  expect(setShapes).toHaveBeenCalledTimes(12);
});
