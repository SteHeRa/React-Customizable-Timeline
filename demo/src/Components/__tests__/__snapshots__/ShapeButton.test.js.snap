// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShapeButton component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h4>
           
          Title Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="titleShapecircle"
              id="titleShapecircle"
              name="titleShape"
              type="radio"
              value="circle"
            />
            <label
              for="titleShapecircle"
            >
              CIRCLE
            </label>
          </div>
          <div>
            <input
              aria-label="titleShapesquare"
              id="titleShapesquare"
              name="titleShape"
              type="radio"
              value="square"
            />
            <label
              for="titleShapesquare"
            >
              SQUARE
            </label>
          </div>
          <div>
            <input
              aria-label="titleShaperectangular"
              id="titleShaperectangular"
              name="titleShape"
              type="radio"
              value="rectangular"
            />
            <label
              for="titleShaperectangular"
            >
              RECTANGULAR
            </label>
          </div>
          <div>
            <input
              aria-label="titleShapediamond"
              id="titleShapediamond"
              name="titleShape"
              type="radio"
              value="diamond"
            />
            <label
              for="titleShapediamond"
            >
              DIAMOND
            </label>
          </div>
        </div>
      </div>
      <div>
        <h4>
           
          Dot Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="dotShapecircle"
              id="dotShapecircle"
              name="dotShape"
              type="radio"
              value="circle"
            />
            <label
              for="dotShapecircle"
            >
              CIRCLE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapesquare"
              id="dotShapesquare"
              name="dotShape"
              type="radio"
              value="square"
            />
            <label
              for="dotShapesquare"
            >
              SQUARE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapeline"
              id="dotShapeline"
              name="dotShape"
              type="radio"
              value="line"
            />
            <label
              for="dotShapeline"
            >
              LINE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapediamond"
              id="dotShapediamond"
              name="dotShape"
              type="radio"
              value="diamond"
            />
            <label
              for="dotShapediamond"
            >
              DIAMOND
            </label>
          </div>
        </div>
      </div>
      <div>
        <h4>
           
          Line Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="lineShapesolid"
              id="lineShapesolid"
              name="lineShape"
              type="radio"
              value="solid"
            />
            <label
              for="lineShapesolid"
            >
              SOLID
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapedotted"
              id="lineShapedotted"
              name="lineShape"
              type="radio"
              value="dotted"
            />
            <label
              for="lineShapedotted"
            >
              DOTTED
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapedashed"
              id="lineShapedashed"
              name="lineShape"
              type="radio"
              value="dashed"
            />
            <label
              for="lineShapedashed"
            >
              DASHED
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapehidden"
              id="lineShapehidden"
              name="lineShape"
              type="radio"
              value="hidden"
            />
            <label
              for="lineShapehidden"
            >
              HIDDEN
            </label>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <h4>
           
          Title Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="titleShapecircle"
              id="titleShapecircle"
              name="titleShape"
              type="radio"
              value="circle"
            />
            <label
              for="titleShapecircle"
            >
              CIRCLE
            </label>
          </div>
          <div>
            <input
              aria-label="titleShapesquare"
              id="titleShapesquare"
              name="titleShape"
              type="radio"
              value="square"
            />
            <label
              for="titleShapesquare"
            >
              SQUARE
            </label>
          </div>
          <div>
            <input
              aria-label="titleShaperectangular"
              id="titleShaperectangular"
              name="titleShape"
              type="radio"
              value="rectangular"
            />
            <label
              for="titleShaperectangular"
            >
              RECTANGULAR
            </label>
          </div>
          <div>
            <input
              aria-label="titleShapediamond"
              id="titleShapediamond"
              name="titleShape"
              type="radio"
              value="diamond"
            />
            <label
              for="titleShapediamond"
            >
              DIAMOND
            </label>
          </div>
        </div>
      </div>
      <div>
        <h4>
           
          Dot Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="dotShapecircle"
              id="dotShapecircle"
              name="dotShape"
              type="radio"
              value="circle"
            />
            <label
              for="dotShapecircle"
            >
              CIRCLE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapesquare"
              id="dotShapesquare"
              name="dotShape"
              type="radio"
              value="square"
            />
            <label
              for="dotShapesquare"
            >
              SQUARE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapeline"
              id="dotShapeline"
              name="dotShape"
              type="radio"
              value="line"
            />
            <label
              for="dotShapeline"
            >
              LINE
            </label>
          </div>
          <div>
            <input
              aria-label="dotShapediamond"
              id="dotShapediamond"
              name="dotShape"
              type="radio"
              value="diamond"
            />
            <label
              for="dotShapediamond"
            >
              DIAMOND
            </label>
          </div>
        </div>
      </div>
      <div>
        <h4>
           
          Line Shape
           
        </h4>
        <div
          class="shapes_selection"
        >
          <div>
            <input
              aria-label="lineShapesolid"
              id="lineShapesolid"
              name="lineShape"
              type="radio"
              value="solid"
            />
            <label
              for="lineShapesolid"
            >
              SOLID
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapedotted"
              id="lineShapedotted"
              name="lineShape"
              type="radio"
              value="dotted"
            />
            <label
              for="lineShapedotted"
            >
              DOTTED
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapedashed"
              id="lineShapedashed"
              name="lineShape"
              type="radio"
              value="dashed"
            />
            <label
              for="lineShapedashed"
            >
              DASHED
            </label>
          </div>
          <div>
            <input
              aria-label="lineShapehidden"
              id="lineShapehidden"
              name="lineShape"
              type="radio"
              value="hidden"
            />
            <label
              for="lineShapehidden"
            >
              HIDDEN
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h4>
         
        Title Shape
         
      </h4>
      <div
        class="shapes_selection"
      >
        <div>
          <input
            aria-label="titleShapecircle"
            id="titleShapecircle"
            name="titleShape"
            type="radio"
            value="circle"
          />
          <label
            for="titleShapecircle"
          >
            CIRCLE
          </label>
        </div>
        <div>
          <input
            aria-label="titleShapesquare"
            id="titleShapesquare"
            name="titleShape"
            type="radio"
            value="square"
          />
          <label
            for="titleShapesquare"
          >
            SQUARE
          </label>
        </div>
        <div>
          <input
            aria-label="titleShaperectangular"
            id="titleShaperectangular"
            name="titleShape"
            type="radio"
            value="rectangular"
          />
          <label
            for="titleShaperectangular"
          >
            RECTANGULAR
          </label>
        </div>
        <div>
          <input
            aria-label="titleShapediamond"
            id="titleShapediamond"
            name="titleShape"
            type="radio"
            value="diamond"
          />
          <label
            for="titleShapediamond"
          >
            DIAMOND
          </label>
        </div>
      </div>
    </div>
    <div>
      <h4>
         
        Dot Shape
         
      </h4>
      <div
        class="shapes_selection"
      >
        <div>
          <input
            aria-label="dotShapecircle"
            id="dotShapecircle"
            name="dotShape"
            type="radio"
            value="circle"
          />
          <label
            for="dotShapecircle"
          >
            CIRCLE
          </label>
        </div>
        <div>
          <input
            aria-label="dotShapesquare"
            id="dotShapesquare"
            name="dotShape"
            type="radio"
            value="square"
          />
          <label
            for="dotShapesquare"
          >
            SQUARE
          </label>
        </div>
        <div>
          <input
            aria-label="dotShapeline"
            id="dotShapeline"
            name="dotShape"
            type="radio"
            value="line"
          />
          <label
            for="dotShapeline"
          >
            LINE
          </label>
        </div>
        <div>
          <input
            aria-label="dotShapediamond"
            id="dotShapediamond"
            name="dotShape"
            type="radio"
            value="diamond"
          />
          <label
            for="dotShapediamond"
          >
            DIAMOND
          </label>
        </div>
      </div>
    </div>
    <div>
      <h4>
         
        Line Shape
         
      </h4>
      <div
        class="shapes_selection"
      >
        <div>
          <input
            aria-label="lineShapesolid"
            id="lineShapesolid"
            name="lineShape"
            type="radio"
            value="solid"
          />
          <label
            for="lineShapesolid"
          >
            SOLID
          </label>
        </div>
        <div>
          <input
            aria-label="lineShapedotted"
            id="lineShapedotted"
            name="lineShape"
            type="radio"
            value="dotted"
          />
          <label
            for="lineShapedotted"
          >
            DOTTED
          </label>
        </div>
        <div>
          <input
            aria-label="lineShapedashed"
            id="lineShapedashed"
            name="lineShape"
            type="radio"
            value="dashed"
          />
          <label
            for="lineShapedashed"
          >
            DASHED
          </label>
        </div>
        <div>
          <input
            aria-label="lineShapehidden"
            id="lineShapehidden"
            name="lineShape"
            type="radio"
            value="hidden"
          />
          <label
            for="lineShapehidden"
          >
            HIDDEN
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
